backend/src/test/java/com/google/sps/UserDatastoreTest.java

```diff
@@ -0,0 +1,102 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.User;
+import com.google.sps.datastore.UserDatastore;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserDatastoreTest {
+
+  // Default values
+  private static final long ID_DEFAULT = 0;
+
+  // Some usernames
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void addOneUser() {
+    // Add one user to datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserDatastore userDatastore = new UserDatastore(datastore);
+    User userA = new User(ID_DEFAULT, PERSON_A);
+    userDatastore.addUser(userA);
+
+    String expected = "username=" + PERSON_A + "\n";
+    assertThat(userDatastore.toString()).isEqualTo(expected);
```

Comment:
here and below: instead of checking userDataStore.toString, I would recommend accessing `datastore` (the DatasotreService) and manually checking that the expected inserts occurred

Reply:
Got it, changed.